File: Laba4/Java/HierarchyStrategy.java
class HierarchyStrategy implements Strategy {
    @Override
    public void takeForks(int id, Table table) {
        if (id == Main.PHILOSOPHERS_COUNT - 1) {
            table.getFork((id + 1) % Main.PHILOSOPHERS_COUNT);
            table.getFork(id);
        } else {
            table.getFork(id);
            table.getFork((id + 1) % Main.PHILOSOPHERS_COUNT);
        }
    }

    @Override
    public void putForks(int id, Table table) {
        if (id == Main.PHILOSOPHERS_COUNT - 1) {
            table.putFork(id);
            table.putFork((id + 1) % Main.PHILOSOPHERS_COUNT);
        } else {
            table.putFork((id + 1) % Main.PHILOSOPHERS_COUNT);
            table.putFork(id);
        }
    }
}

File: Laba4/Sharp/HierarchyStrategy.cs
namespace Laba4
{
    public class HierarchyStrategy : IStrategy
    {
        public void TakeForks(int id, Table table)
        {
            if (id == Program.PhilosophersCount - 1)
            {
                table.GetFork((id + 1) % Program.PhilosophersCount);
                table.GetFork(id);
            }
            else
            {
                table.GetFork(id);
                table.GetFork((id + 1) % Program.PhilosophersCount);
            }
        }

        public void PutForks(int id, Table table)
        {
            if (id == Program.PhilosophersCount - 1)
            {
                table.PutFork(id);
                table.PutFork((id + 1) % Program.PhilosophersCount);
            }
            else
            {
                table.PutFork((id + 1) % Program.PhilosophersCount);
                table.PutFork(id);
            }
        }
    }
}